clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -1303,11 +1305,13 @@ public void testCommitWithTopicAuthorizationFailureInAddPartitionsInFlight() thr
         Map<TopicPartition, Errors> errors = new HashMap<>();
         errors.put(tp0, Errors.TOPIC_AUTHORIZATION_FAILED);
         errors.put(tp1, Errors.OPERATION_NOT_ATTEMPTED);
+        AddPartitionsToTxnResult result = AddPartitionsToTxnResponse.resultForTransaction(AddPartitionsToTxnResponse.V3_AND_BELOW_TXN_ID, errors);
+        AddPartitionsToTxnResponseData data = new AddPartitionsToTxnResponseData().setResultsByTopicV3AndBelow(result.topicResults()).setThrottleTimeMs(0);
         client.respond(body -> {
             AddPartitionsToTxnRequest request = (AddPartitionsToTxnRequest) body;
-            assertEquals(new HashSet<>(request.partitions()), new HashSet<>(errors.keySet()));
+            assertEquals(new HashSet<>(AddPartitionsToTxnRequest.getPartitions(request.data().v3AndBelowTopics())), new HashSet<>(errors.keySet()));
```

Comment:
I tend to agree with this. Are those helpers only used in tests? If so, it may be better to put them in `AddPartitionsToTxnRequestTest`. What do you think?

Reply:
I can take a look. I also used internally to build the request and I think the v4+ version in KafkaApis itself. I can try to make a helper for these tests though.

Reply:
If those are used internally the request as well, we can't really move them.

Reply:
The one in the request is used in a place that iterates through the topic objects so it is actually ok. I made a helper in the TransactionManagerTest since it uses the phrase 3 times. I left the final remaining usage of this in the last test file since I think its ok to have once and doesn't make sense to use a helper.